core/src/main/java/org/apache/iceberg/V2Metadata.java

```diff
@@ -243,11 +243,11 @@ static Schema wrapFileSchema(Types.StructType fileSchema) {
 
   static class IndexedManifestEntry implements ManifestEntry, IndexedRecord {
     private final org.apache.avro.Schema avroSchema;
-    private final long commitSnapshotId;
+    private final Long commitSnapshotId;
```

Comment:
I think we have to update the condition in `get` as `commitSnapshotId` is now `Long` and we cannot use `==`.

Reply:
Isn't this okay because it will unbox the values first? Using `equals` would duplicate the null checks, I thought.

Reply:
Fixed